server/parameters/util/OptionParser.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.vaticle.typedb.core.server.parameters.util;
+
+import com.vaticle.typedb.core.common.exception.TypeDBException;
+import com.vaticle.typedb.core.common.iterator.FunctionalIterator;
+
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Server.CLI_FLAG_OPTION_HAS_VALUE;
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Server.CLI_OPTION_REQUIRES_TYPED_VALUE;
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Server.DUPLICATE_CLI_OPTION;
+import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
+import static com.vaticle.typedb.core.server.parameters.util.YamlParser.concatenate;
+
+public abstract class OptionParser {
+
+    public final java.lang.String name;
+    private final java.lang.String description;
+    private final java.lang.String typeDescription;
+
+    public OptionParser(java.lang.String name, java.lang.String description, java.lang.String typeDescription) {
+        this.name = name;
+        this.description = description;
+        this.typeDescription = typeDescription;
+    }
+
+    public java.lang.String name() {
+        return name;
+    }
+
+    java.lang.String description() {
+        return description;
+    }
+
+    public HelpEntry helpEntry(java.lang.String optionScope) {
```

Comment:
I think since this is no longer implementing the same interface, it doesn't make sense to have an `optionScope` here - all options are the root scope of the arg parser right?